File: backend/src/models/XetNghiem.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/sequelize');

const XetNghiem = sequelize.define('XetNghiem', {
  maXN: {
    type: DataTypes.STRING(100),
    primaryKey: true
  },
  maLoaiXN: {
    type: DataTypes.STRING(100),
    allowNull: false
  },
  tenXN: {
    type: DataTypes.STRING(100),
    allowNull: false
  },
  chiPhi: {
    type: DataTypes.DECIMAL(12, 2),
    allowNull: false
  },
  thoiGianTraKetQua: {
    type: DataTypes.STRING(100)
  }
}, {
  tableName: 'XetNghiem',
  timestamps: false
});

XetNghiem.associate = (models) => {
  XetNghiem.belongsTo(models.LoaiXetNghiem, { foreignKey: 'maLoaiXN' });
};

module.exports = XetNghiem;

File: FrontendApp/app/src/main/java/com/example/frontendapp/data/model/test/XetNghiem.kt
package com.example.frontendapp.data.model.test

data class XetNghiem(
    val maXN: String,
    val maLoaiXN: String,
    val tenXN: String,
    val chiPhi: Double,
    val thoiGianTraKetQua: String?,
    val loaiXetNghiem: LoaiXetNghiem? = null   // from belongsTo
)
